src/components/functional_module/src/plugin_manager.cc

```diff
@@ -43,6 +43,38 @@ namespace functional_modules {
 
 CREATE_LOGGERPTR_GLOBAL(logger_, "PluginManager")
 
+namespace {
+const std::string ExtractMethodName(application_manager::MessagePtr msg) {
+  Json::Value value;
+  Json::Reader reader;
+  reader.parse(msg->json_message(), value);
+  
+  const char* kMethod = "method";
+  const char* kResult = "result";
+  const char* kError = "error";
+  const char* kData = "data";
+
+  if (value.isMember(kMethod)) {
+    return value.get(kMethod, "").asCString();
+    // Response from HMI
+  }
+  if (value.isMember(kResult)) {
+    const Json::Value& result = value.get(kResult, Json::Value());
+    return result.get(kMethod, "").asCString();
+  }
+  if (value.isMember(kError)) {
+    const Json::Value& error = value.get(kError, Json::Value());
+    const Json::Value& data = error.get(kData, Json::Value());
+    return data.get(kMethod, "").asCString();
+  }
+  LOG4CXX_WARN(logger_,
+               "Message with HMI protocol version can not be handled by "
```

Comment:
@LuxoftAKutsan 
This method was moved from another place 'as is'. Will change it you want.